File: MIDTERM-LAB-1/JAVA/DataRecord.java
import java.util.Date;

// Class to represent a single data record from the dataset
public class DataRecord {
    private Date date;          // Release date of the record
    private double sales;       // Sales value
    private Double movingAvg;   // 3-day moving average (nullable)

    // Constructor initializes date and sales
    public DataRecord(Date date, double sales) {
        this.date = date;
        this.sales = sales;
    }

    // Getter for date
    public Date getDate() { return date; }

    // Getter for sales
    public double getSales() { return sales; }

    // Getter for moving average
    public Double getMovingAvg() { return movingAvg; }

    // Setter for moving average
    public void setMovingAvg(Double movingAvg) { this.movingAvg = movingAvg; }
}

File: MIDTERM-LAB-1/JAVASCRIPT/dataRecord.js
class DataRecord {
  constructor(date, sales) {
    this.date = date;          // Release date
    this.sales = sales;        // Sales value
    this.movingAvg = null;     // 3-day moving average (nullable)
  }

  setMovingAvg(avg) {
    this.movingAvg = avg;
  }

  getDate() {
    return this.date;
  }

  getSales() {
    return this.sales;
  }

  getMovingAvg() {
    return this.movingAvg;
  }
}

// Function to compute 3-day moving average
function computeMovingAverage(records) {
  for (let i = 0; i < records.length; i++) {
    if (i < 2) {
      records[i].setMovingAvg(null);
    } else {
      const avg = (
        records[i].getSales() +
        records[i - 1].getSales() +
        records[i - 2].getSales()
      ) / 3;
      records[i].setMovingAvg(avg);
    }
  }
}

module.exports = { DataRecord, computeMovingAverage };
